src/api/Nhs.Appointments.Api/Functions/GetWeekSummaryFunction.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using FluentValidation;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Attributes;
+using Microsoft.Extensions.Logging;
+using Microsoft.OpenApi.Models;
+using Nhs.Appointments.Api.Auth;
+using Nhs.Appointments.Api.Models;
+using Nhs.Appointments.Core;
+using Nhs.Appointments.Core.Features;
+using Nhs.Appointments.Core.Inspectors;
+
+namespace Nhs.Appointments.Api.Functions;
+
+public class GetWeekSummaryFunction(
+    IBookingAvailabilityStateService bookingAvailabilityStateService,
+    IFeatureToggleHelper featureToggleHelper,
+    IValidator<GetWeekSummaryRequest> validator,
+    IUserContextProvider userContextProvider,
+    ILogger<GetWeekSummaryFunction> logger,
+    IMetricsRecorder metricsRecorder)
+    : BaseApiFunction<GetWeekSummaryRequest, WeekSummary>(validator, userContextProvider,
+        logger, metricsRecorder)
+{
+    [OpenApiOperation(operationId: "GetWeekSummary", tags: ["Availability"],
+        Summary = "Get weekly availability summary for a week start date and site")]
+    [OpenApiParameter("site", In = ParameterLocation.Query, Required = true, Type = typeof(double),
+        Description = "The id of them site where to get daily availability from")]
```

Comment:
Very minor typo in "them" :)

Also, apologies for being a massive pedant but since I'm commenting on this line anyway could we clean up the dangling participle? Perhaps "The ID of the site from which to query bookings and availability"? ;)